### DISCUSSION TITLE
Directory - Standalone/deployments

### DISCUSSION THREAD

**Discussion by Jerry161984** [keycloak/keycloak#12708] [Q&A]
Hi Keycloak Families,

I recently started to work as Associate IS Security officer and I have been given to work in Identity and Access Management system with keycloak. I did the work by reading the documentation in the Keycloak website. I have an issue with the customization of the Authentication part;

We are trying to implement MFA for our organization and we are considering to customize keycloak Email OTP and SMS OTP. On the previous Versions (V15.1.1-V16) of Keycloak there was a Directory called **standalone/deployments** to deploy the **.jar** files for the features to be added in the Runtime environment.  For V 15.1.1 it works great.

I am currently using **Keycloak Version 18.0.1** for doing the MFA job. Within this version there is no standalone directory for the .jar files to be deployed. Can anyone help me how to customize this MFA authentication?

Thanks for the help!
Cheers :)
![Version 15](https://user-images.githubusercontent.com/66791123/175486811-17b2f577-61bb-4ed6-b057-1422c39f9bd2.png)
![Version 18](https://user-images.githubusercontent.com/66791123/175486820-5599be41-87b0-4db7-b712-babcd8599f56.png)

**Comment by nikitaron** [comment]
Hi! Faced the same question, maybe you already found the answer?

**Comment by darius-m** [comment]
I think the jars should go into the `providers` directory - https://www.keycloak.org/server/configuration-provider.
